app/playground/page.client.tsx

```diff
@@ -0,0 +1,48 @@
+'use client';
+
+import { Alert, AlertIcon, Center, Flex, Heading } from '@chakra-ui/react';
+
+interface PageClientProps {
+  posts: {
+    id: number;
+    title: string;
+    author: string;
+  }[];
+}
+
+const PageClient = ({ posts }: PageClientProps) => (
+  <Center width="100%">
+    <Flex flexDirection="column" gap="24px" width="100%" maxWidth="1500px">
+      <Heading
+        as="h1"
+        size={{ base: 'md', md: 'lg', lg: 'lg' }}
+        marginX={{ base: 'auto', md: '0', lg: '0' }}
+      >
+        Playground
+      </Heading>
+      Developer playground for examples of component or feature use.
+      <Heading
+        as="h2"
+        size={{ base: 'xs', md: 'sm', lg: 'md' }}
+        marginX={{ base: 'auto', md: '0', lg: '0' }}
+      >
+        Server Side Data Fetch
+      </Heading>
+      {posts ? (
+        posts.map((item) => (
+          <div key={item.id}>
+            <p>Post Title: {item.title}</p>
+            <p>Author: {item.author}</p>
+          </div>
+        ))
+      ) : (
+        <Alert status="error">
+          <AlertIcon />
+          Failed to fetch post data
+        </Alert>
+      )}
```

Comment:
Minor syntax point - should this ternary not be refactored with a [nullish coalescence operator?](https://github.com/airbnb/javascript#comparison--unneeded-ternary)

Reply:
Not sure how that can be applied here, I need to check for presence of posts and render different components based on that. Can you show an example? I mean I can avoid using ternary by making a function or something like this 

```
function Posts({posts}) {
  if(!posts) {
    return (
      <Alert status="error">
        <AlertIcon />
        Failed to fetch post data
      </Alert>
    );
  }

  return posts.map(post => <Post key={post.id} post={post} />);
}
```

Reply:
```
{posts?.map((item) => (
  <div key={item.id}>
  <p>Post Title: {item.title}</p>
  <p>Author: {item.author}</p>
  </div>
)) ?? (
  <Alert status="error">
    <AlertIcon />
      Failed to fetch post data
    </Alert>
)}
```

Reply:
Damn, just posted a commit with everything else covered. Sending another commit